agent_code/our_agent_v2/train.py

```diff
@@ -210,11 +243,12 @@ def end_of_round(self, last_game_state: dict, last_action: str, events: List[str
         raise ValueError(f"Unknown act_strategy {self.act_strategy}")
 
     # ---------- (3) TD Q-learning with batch: ----------
+    
     if len(self.transitions) > BATCH_SIZE and self.game_nr % TRAIN_FREQ == 0:
         # Create a random batch from the transition history.
         batch = random.sample(self.transitions, BATCH_SIZE)
         X, targets = [], []
-        for state, action, state_next, reward in batch:
+        for state, action, state_next, n_step_reward, n_step_next_state in batch:
```

Comment:
Since state_next is not needed we can probably remove this, and also not store this is in the transition deque

Reply:
You are right, however I wasn't sure you need it at any point so I didn't want to delete it - if not I will